### PR TITLE
Allow status tracker to send commit status to a GitLab repository

### PR DESCRIPTION
**Pull request by chetan-rns**
# Changes

This PR will provide support for the operator to send the commit status to a GitLab repository. This involves detecting the repository type and using its client.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

### CODE REVIEW

File: commit-status-tracker/pkg/controller/pipelinerun/pipelinerun_controller_test.go [hunk 1]
```diff
@@ -354,7 +354,7 @@ func makeReconciler(pr *pipelinev1.PipelineRun, objs ...runtime.Object) (*Reconc
 	s.AddKnownTypes(pipelinev1.SchemeGroupVersion, pr)
 	cl := fake.NewFakeClient(objs...)
 	client, data := fakescm.NewDefault()
-	fakeClientFactory := func(s string) *scm.Client {
+	fakeClientFactory := func(s, repoType string) *scm.Client {
```
Comment (bigkevmcd) [inline commit-status-tracker/pkg/controller/pipelinerun/pipelinerun_controller_test.go:357]: One thing you could do here, is assert that `repoType` matches what you expect, and fail the test if not?

File: commit-status-tracker/pkg/controller/pipelinerun/client.go [hunk 2]
```diff
@@ -19,14 +19,21 @@ import (
...
 
-type scmClientFactory func(string) *scm.Client
+type scmClientFactory func(string, string) *scm.Client
 
-// TODO: fix this to determine the type of scm Client to create.
-func createClient(token string) *scm.Client {
-	client := github.NewDefault()
+func createClient(token, repoType string) *scm.Client {
+	var client *scm.Client
+	if repoType == "github" {
```
Comment (chmouel) [inline commit-status-tracker/pkg/controller/pipelinerun/client.go:30]: I think you could add api.github to support github enteprise here...

**Pull request merged**